src/modbus-ucl.h

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright © 2018 Lars Täuber <gitaeuber@users.noreply.github.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#ifndef MODBUS_UCL_H
+#define MODBUS_UCL_H
+
+MODBUS_BEGIN_DECLS
+
+const char *ucl_msg_type[] = {
```

Comment:
Do not put a variable definition in a .h file. It will be included in all .c-file including it and then, at link time, you'll get "multiple definitions" as can be seen in your output.